File: src/main/java/org/noxet/noxetserver/combatlogging/CombatLoggingStorageManager.java
package org.noxet.noxetserver.combatlogging;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.util.ConfigManager;
import org.noxet.noxetserver.playerstate.PlayerState;
import org.noxet.noxetserver.realm.RealmManager;

import java.util.List;

public class CombatLoggingStorageManager extends ConfigManager {
    @Override
    protected String getFileName() {
        return "combat-logging";
    }

    public void setPlayerCombatLogMode(Player player, RealmManager.Realm realm, boolean combatLogged) {
        String key = (realm != null ? realm.getPlayerStateType() : PlayerState.PlayerStateType.GLOBAL).name();

        List<String> realmCombatList = config.getStringList(key);

        if(combatLogged)
            realmCombatList.add(player.getUniqueId().toString());
        else
            realmCombatList.remove(player.getUniqueId().toString());

        config.set(key, realmCombatList);
        save();
    }

    public boolean isCombatLogged(Player player, RealmManager.Realm realm) {
        String key = (realm != null ? realm.getPlayerStateType() : PlayerState.PlayerStateType.GLOBAL).name();
        return config.getStringList(key).contains(player.getUniqueId().toString());
    }

    public void combatLogRejoin(Player player, RealmManager.Realm realm) {
        if(isCombatLogged(player, realm)) { // Check if player left while combat logged last time.
            setPlayerCombatLogMode(player, realm, false); // Reset combat log state.
            player.setHealth(0); // Kill player!
        }
    }
}

File: src/main/java/org/noxet/noxetserver/combatlogging/CombatLoggingStorageManager.scala
package org.noxet.noxetserver.combatlogging

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerState
import org.noxet.noxetserver.realm.RealmManager.Realm
import org.noxet.noxetserver.util.ConfigManager

class CombatLoggingStorageManager extends ConfigManager:
  override def getFileName = "combat-logging"

  def setPlayerCombatLogMode(player: Player, realm: Realm, combatLogged: Boolean): Unit =
    val key = (
      if realm != null then realm.getPlayerStateType
      else PlayerState.PlayerStateType.GLOBAL
    ).name

    val realmCombatList = config.getStringList(key)

    if combatLogged then
      realmCombatList.add(player.getUniqueId.toString)
    else
      realmCombatList.remove(player.getUniqueId.toString)

    config.set(key, realmCombatList)
    save()

  def isCombatLogged(player: Player, realm: Realm): Boolean =
    val key = (
      if realm != null then realm.getPlayerStateType
      else PlayerState.PlayerStateType.GLOBAL
    ).name
    config.getStringList(key).contains(player.getUniqueId.toString)

  def triggerRejoin(player: Player, realm: Realm): Unit =
    if isCombatLogged(player, realm) then // Check if player left while combat logged last time.
      setPlayerCombatLogMode(player, realm, false) // Reset combat log state.
      player.setHealth(0) // Kill player!